### PR TITLE
Toggle search result filters for mobile

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
The search results filters take up too much space on mobile

### Changes proposed in this pull request
Hide them behind toggle button

### Guidance to review
Once on the results page on mobile click "Filter the results" and the filters should become visible.

#Mobile After
screen shot 2018-08-06 at 14 10 58
screen shot 2018-08-06 at 14 11 03

# Change on desktop 
localhost_5000_results_l 2 subjects 2 funding 15 pgce false qts false fulltime false parttime false

### CODE REVIEW

File: src/Assets/Styles/components/_toggle.scss [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  border: none;
+  color: $govuk-text-colour;
+  display: none;
+  padding: 8px;
+  text-align: left;
+  text-decoration: none;
+  width: 100%;
+
+  &.is-toggled {
+    background-position: right -75px;
```
Comment (fofr) [inline src/Assets/Styles/components/_toggle.scss]: The arrow should be the other way around, the down arrow should show when the filter list is closed. This is what GOV.UK does: https://www.gov.uk/search?q=test

screen shot 2018-08-06 at 15 30 16

File: src/Assets/Javascript/toggle.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Toggle element visibility
+var toggle = function (elem) {
+  elem.classList.toggle('is-visible');
+};
+
+// Listen for click events
+document.addEventListener('click', function (event) {
```
Comment (fofr) [inline src/Assets/Javascript/toggle.js]: Can the JS also add `aria` attributes when it first executes, and then update them when toggled to represent state?

File: src/Assets/Styles/components/_toggle.scss [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.js-toggle-content {
+  display: block;
+
+  .js-enabled & {
+    display: none;
+
+    // always show for desktop
+    @include mq ($from: desktop) {
```
Comment (fofr) [inline src/Assets/Styles/components/_toggle.scss]: Does this mean that any browser that doesn't support media queries won't see filters? Probably only IE8

Comment (dankmitchell) [reply src/Assets/Styles/components/_toggle.scss]: Correct. Do we support IE8? https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices

File: src/Views/Results/Index.cshtml [hunk 4]
```diff
@@ -16,7 +16,7 @@
   </div>
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-one-third">
-      <button data-target="#searchFilters" class="js-toggle">
+      <button data-target="#searchFilters" class="js-toggle" aria-expanded="false">
```
Comment (fofr) [inline src/Views/Results/Index.cshtml]: Shouldn't the `aria-expanded` be on the target, then this should have `aria-controls`?

Comment (dankmitchell) [reply src/Views/Results/Index.cshtml]: No, `aria-expanded` should be on the trigger as per https://www.w3.org/WAI/GL/wiki/Using_aria-expanded_to_indicate_the_state_of_a_collapsible_element#Description. I can definitely add an `aria-controls` attribute though 👌

**Pull request merged**